tests/polkadot_tests/calls.rs

```diff
@@ -944,102 +938,68 @@ contract Flagger {
     // Yields different result for tail calls
     function tail_call_it(
         address _address,
-        bytes4 _selector,
         uint32 _x,
-        CallFlag[] _flags
+        uint32 _flags
     ) public returns(uint32 ret) {
-        bytes input = abi.encode(_selector, _x);
-        (bool ok, bytes raw) =  _address.call{flags: bitflags(_flags)}(input);
+        bytes input = abi.encode(bytes4(0), _x);
+        (bool ok, bytes raw) =  _address.call{flags: _flags}(input);
         require(ok);
         ret = abi.decode(raw, (uint32));
         ret += 1;
     }
-}"##,
-    );
+}"##;
 
-    #[derive(Encode)]
-    enum CallFlags {
-        ForwardInput,
-        CloneInput,
-        TailCall,
-        AllowReentry,
-    }
-    #[derive(Encode)]
-    struct Input {
-        address: [u8; 32],
-        selector: [u8; 4],
-        voyager: u32,
-        flags: Vec<CallFlags>,
-    }
+    let forward_input = 0b1u32;
+    let clone_input = 0b10u32;
+    let tail_call = 0b100u32;
+    let allow_reentry = 0b1000u32;
```

Comment:
For clarity, you can write these as `1`, `1 << 1`, `1 << 2` and `1 << 3`.

Reply:
I don't think `1 << 3` is necessarily clearer than `0b100`.

Reply:
My reasoning was that I don't need to count the zeros to the right of `1` when I use shift, because `1 << 2` shows me exactly the set bit. 

Either way, it is up to Cyrill whether to change or not.